plugins/inputs/logfile/tail/tail.go

```diff
@@ -182,11 +187,25 @@ func (tail *Tail) closeFile() {
 
 func (tail *Tail) reopen() error {
 	tail.closeFile()
+	numTries := 0
+	waitDuration := fileOpenRetryDuration
 	for {
 		var err error
 		tail.file, err = OpenFile(tail.Filename)
 		tail.curOffset = 0
 		if err != nil {
+			if os.IsPermission(err) {
+				// Specifically on Windows, reopening the file too quickly can result in an access denied error.
+				// Retry to allow for some leniency for when the file on the host becomes available.
+				tail.Logger.Debugf("Access denied on %s. Retried %d times so far", tail.Filename, numTries)
+				if numTries < fileOpenMaxRetries {
+					tail.Logger.Debugf("Sleeping for %v ms and retrying", waitDuration.Milliseconds())
+					time.Sleep(waitDuration)
+				}
+				numTries += 1
+				waitDuration *= 2
+				continue
+			}
 			if os.IsNotExist(err) {
 				tail.Logger.Debugf("Waiting for %s to appear...", tail.Filename)
 				if err := tail.watcher.BlockUntilExists(&tail.Tomb); err != nil {
```

Comment:
So does this mean the `tail` never dies now?
If the user is monitoring `/tmp/foo*.log` and continuously creates files like:
```
foo1.log
foo2.log
foo3.log
...
```
Even if the old files get deleted will the `tail` live on, stuck in `BlockUntilExists()`?

Reply:
I just noticed this myself and am fixing this

Reply:
Let me back up

> Even if the old files get deleted will the tail live on, stuck in BlockUntilExists()

I was concerned about this myself. That was partly why I had the last set of metrics I provided in the PR description.
![Screen Shot 2022-04-27 at 10 02 21 AM](https://user-images.githubusercontent.com/17056456/165536603-ba3e84f2-0875-4530-a355-510ab607acf9.png)

Let me pull debug logs from both of those hosts

---
What I was referring to "fixing" just now is that I messed up the retry on file permission errors.

Reply:
I would expect without `auto_removal` that this function would wait forever for the file to get recreated.

Reply:
That's what I expect too. The question is the impact. Will repurpose some hosts I used for this to see what happens when I constantly rotate to a new file.

Reply:
What I'm doing now for validating is I'm using this script to generate 10,000 log files:
```
import logging
import time

logger = logging.getLogger('test')
logger.setLevel(logging.INFO)

def write_batch(logger, batch):
    for i in range(100):
        logger.info("This is the %dth batch: %d", batch, i)
        time.sleep(0.001)

for batch in range(1, 10001):
    file_name = 'foo.log.'+str(batch)
    fh = logging.FileHandler(file_name)
    logger.addHandler(fh)
    write_batch(logger, batch)
    logger.removeHandler(fh)
    time.sleep(0.1)
```
I set auto_removal to false on both test hosts.

After all of the log files get written, CPU/memory spike up on both hosts, which makes sense because the agent is now tailing 10k log files, though the log files aren't being updated after the script finishes.

Now, I'm going through and deleting some of the log files, like `rm foo.log.1*` to delete 1000 of the log files, and monitoring the CPU/mem usage as I remove log files to monitor. Basically, I want to see how drastically they diverge in held CPU and memory when I delete log files that it tries to constantly reopen. 

Will post my findings here as I get more data.

Reply:
![Screen Shot 2022-04-27 at 2 53 07 PM](https://user-images.githubusercontent.com/17056456/165598949-e008b7ca-2cd3-49e3-968f-1efe2fa0de22.png)
I'm not very happy with my findings. Only sharing the process CPU usage because the other metrics align pretty closely despite the change in behavior. This is something I would want to drill down on before merging this change. 

Something worth noting - as I was looking at existing unit test behavior, I saw that there is a timer for cleaning up a tail process if the file is closed. This is controlled by the `exitOnDeletion()` function. 

This is currently, with my change here, not executed. 
```go
	if !config.ReOpen {
		go t.exitOnDeletion()
	}
```
I'm thinking that based on the existing durations for this monitoring, which maxes out at 5 minutes, I think that it is safe to remove the `if` check here and just always run this goroutine. I'll make that change and subsequent unit test changes, and then redo this test.

Reply:
Forgot to update on this: what happens is the `exitOnDeletion()` never detects that the file is deleted, so this does nothing to resolve the issue. At this point, I feel like this was wasted effort and I need to pivot. I have test hosts and monitoring set up so it should be quick to get stuff set up to monitor for leaks in whatever other solution I come up with, but unfortunately for @akozd , I don't think this is a viable solution without serious degradation.